src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -312,6 +312,13 @@ private TCommandResult ProcessResponse(ConnectionId connectionId, CommandMessage
                     if (materializedDocument.TryGetValue("errmsg", out errmsgBsonValue) && errmsgBsonValue.IsString)
                     {
                         var errmsg = errmsgBsonValue.ToString();
+                        if (materializedDocument.TryGetValue("code", out var errorCode)
+                            && errorCode.AsInt32 == 20
+                            && errmsg.StartsWith("Transaction numbers"))
+                        {
+                            errmsg = "This MongoDB deployment does not support retryable writes. "
+                                     + "Please add retryWrites=false to your connection string.";
```

Comment:
Weird indenting. I would have written:

    errmsg = 
        "This MongoDB deployment does not support retryable writes. " +
        "Please add retryWrites=false to your connection string.";

Reply:
Fixed!